app/src/main/java/fr/free/nrw/commons/contributions/ContributionController.java

```diff
@@ -46,11 +55,112 @@ public void initiateCameraPick(Activity activity) {
 
         PermissionUtils.checkPermissionsAndPerformAction(activity,
                 Manifest.permission.WRITE_EXTERNAL_STORAGE,
-                () -> initiateCameraUpload(activity),
+                () -> {
+                    if (!(PermissionUtils.hasPermission(activity, Manifest.permission.ACCESS_FINE_LOCATION)
+                         && isLocationAccessToAppsTurnedOn())) {
```

Comment:
Having a check like this to trigger the consent dialog could be a bit incorrect. We attach the location to the upload when the EXIF location is unavailable. This could also occur in cases where the app has location access and location is turned on the device but saving location is _turned off_ in the device camera. The user could've done this intentionally. The app obtained location would be uploaded in this case without the user having any idea about it. I believe it is ideal to get the consent from the user before doing so.

There's also the case of lower Android version devices (such as API 21) for which there is no notion of location permission. So, this dialog is shown to them only when location is turned off.

I also observed this dialog appears each time when trying to use the in-app camera and location is either off / location permissions is unavailable. This might become a bit annoying if the user has intentionally not granted those.

To better handle this, I would suggested showcasing the consent dialog first time (unconditinally) when user users in-app camera and store their preference. Given this would be a one-time dialog, we should also have a corresponding preference in settings in case the user wants to toggle their choice.

When we have user's consent to attach the location, we could then show dialog about lack of location permission / location enabled when in-app camera is triggered.

This seems like a better and reasonable flow to me. Thoughts ?

Reply:
@sivaraam Just a clarification - 

> Given this would be a one-time dialog, we should also have a corresponding preference in settings in case the user wants to toggle their choice.

Is this referring to displaying the consent dialog? 
If yes, I'm a little skeptical about having a setting to toggle the choice to display the dialog. In most apps, we usually have an option like "Do not show this message again" and once the user selects the option, we do not see it again. A user who's curious about enabling location could probably go through the information we have already associated while enabling/disabling permissions for location and would not want a pop up enabled again. 

I may be wrong but just wanted to understand the perspective behind it.

Reply:
> Having a check like this to trigger the consent dialog could be a bit incorrect.

Appeared incorrect to me as well, I felt I was missing out some case but did not know which one. Thank you for elaborating about it, I'll make this dialog compulsory on the run.

Also, I'd suggested this "Do not show this message again" in my comment above. If this sounds okay, I'll add it for the _Dismiss_ case (it does not appear again if the user allows).